pkg/controller/jaeger/upgrade.go

```diff
@@ -18,6 +18,11 @@ func (r *ReconcileJaeger) applyUpgrades(ctx context.Context, jaeger v1.Jaeger) (
 
 	currentVersions := version.Get()
 
+	err := upgrade.AddLabelsToExistingServiceAccounts(ctx, r.rClient, jaeger)
```

Comment:
This applies to all the upgrades. Instead of that, we need to check from what version we're adding the labels to the Service Accounts and create one upgrade routine for that version.

Reply:
This was being checked from version `0.11.0` and the latest version is `1.46.0`. I don't think any one will be using version before a stable one which should be `1.0.0`. So I think this should apply all the future releases.

Reply:
I agree with the first statement. Maybe we can just add this change to one of the previous versions. I would not like to do this every time because users that are not affected by this, will have their service accounts modified.

Even, I would maybe just document the problem and close the issue.

Reply:
`I would not like to do this every time because users that are not affected by this, will have their service accounts modified.`
In the above case, I think documenting the issue and ask users to have a service account name in Jaeger should not same as their existing service account names.
What do you think and also if we decided to document this where we should documents this, I mean is there any doc file where special cases or prerequisites are mentioned?

Reply:
Since we don't have any special documentation for the known issues and others, I would add the final explanation to the issue as a comment. Maybe, ask in https://github.com/jaegertracing/documentation to add a section in the K8s Operator section to document this kind of issues.

Reply:
I don't see any K8s Operator section in [https://github.com/jaegertracing/documentation](https://github.com/jaegertracing/documentation), These documents are mostly related to Jaeger Tracing website.
I think as you suggested put a comment to the issue in detail and close the issue.

Reply:
https://github.com/jaegertracing/documentation/blob/main/content/docs/1.47/operator.md

Reply:
I have created a new [issue](https://github.com/jaegertracing/documentation/issues/664) in the [documentation](https://github.com/jaegertracing/documentation) repo to address this issue.
Now we can close this PR and the issue as well.

Reply:
> I have created a new [issue](https://github.com/jaegertracing/documentation/issues/664) in the [documentation](https://github.com/jaegertracing/documentation) repo to address this issue. Now we can close this PR and the issue as well. @iblancasa , your opinion please?

Hi @iblancasa ,
Please respond on this so that this issue and PR can be closed asap.

Reply:
Ok, let's close the PR and the issue.